nexus_constructor/geometry/disk_chopper_geometry.py

```diff
@@ -0,0 +1,497 @@
+from math import pi
+
+from PySide2.QtGui import QVector3D
+from PySide2.QtWidgets import QListWidget
+import numpy as np
+from numpy import diff, unique
+from numpy.core.umath import deg2rad, ndarray
+
+from nexus_constructor.geometry import OFFGeometryNoNexus
+from nexus_constructor.validators import DATASET_TYPE
+
+SLIT_EDGES = "slit_edges"
+SLITS = "slits"
+RADIUS = "radius"
+SLIT_HEIGHT = "slit_height"
+
+REQUIRED_CHOPPER_FIELDS = [SLIT_EDGES, SLITS, RADIUS, SLIT_HEIGHT]
+CONTAINS_INTEGER = [
+    DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Integer" in key
+]
+CONTAINS_SHORT = [DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Short" in key]
+CONTAINS_LONG = [DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Long" in key]
+INT_TYPES = CONTAINS_INTEGER + CONTAINS_SHORT + CONTAINS_LONG
+
+FLOAT_TYPES = [
+    DATASET_TYPE[key] for key in DATASET_TYPE.keys() if key in ["Float", "Double"]
+]
+
+
+class ChopperChecker:
+    def __init__(self, fields_widget: QListWidget):
```

Comment:
I think some refactoring will be needed here. Probably `ChopperChecker` shouldn't know about the field widget because this validation also needs to be carried out when loading a NeXus file, not just when using the Add Component window.